{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "68f0d299",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "date\n",
    "\n",
    "########################################\n",
    "##### need to specify ################## \n",
    "ecoli_dir=/path/to/save/ecoli/fasta/\n",
    "projPath=/path/to/save/project/\n",
    "cores=4\n",
    "\n",
    "histNames=(C_H3K27me3 C_H3K4me3 C_IgG \\\n",
    "           M_H3K27me3 M_H3K4me3 M_IgG \\\n",
    "           P_H3K27me3 P_H3K4me3 P_IgG)\n",
    "\n",
    "########################################\n",
    "########################################\n",
    "\n",
    "\n",
    "##################################################################\n",
    "## Part 1. Download ecoli genome reference\n",
    "##################################################################\n",
    "ecoli_bowtie2Index=${ecoli_dir}/bowtie2Index/\n",
    "\n",
    "mkdir -p ${ecoli_bowtie2Index}\n",
    "\n",
    "# https://www.ebi.ac.uk/ena/browser/view/U00096\n",
    "wget https://www.ebi.ac.uk/ena/browser/api/fasta/U00096.3 -P ${ecoli_dir} > ${ecoli_dir}/download.log 2>&1\n",
    "\n",
    "##################################################################\n",
    "## Part 2. Build Bowtie2 index for U00096.3 / K12-MG1655\n",
    "##################################################################\n",
    "\n",
    "bowtie2-build --thread ${cores} ${ecoli_dir}/U00096.3 ${ecoli_bowtie2Index}/ecoli_U00096.3 > ${ecoli_bowtie2Index}/build.log 2>&1\n",
    "\n",
    "##################################################################\n",
    "## Part 3 Alignment to spike-in genome\n",
    "##################################################################\n",
    "trimmedPath=${projPath}/processed_data/CUT_RUN/Part1_alignment/\n",
    "outPath=${projPath}/processed_data/CUT_RUN/Part2_ecoli_alignment/\n",
    "\n",
    "mkdir -p ${outPath}/alignment/sam/bowtie2_summary\n",
    "mkdir -p ${outPath}/alignment/bam\n",
    "mkdir -p ${outPath}/alignment/bed\n",
    "mkdir -p ${outPath}/alignment/bedgraph\n",
    "\n",
    "# Loop through each histone marker name\n",
    "for histName in \"${histNames[@]}\"\n",
    "do\n",
    "bowtie2 --very-sensitive --local --very-sensitive-local --no-unal --no-mixed --no-discordant \\\n",
    "        --phred33 \\\n",
    "        -I 10 -X 700 -p ${cores} -x ${ecoli_bowtie2Index}/ecoli_U00096.3 \\\n",
    "        -1 ${trimmedPath}/alignment/trimming/*${histName}*_R1*.fq.gz \\\n",
    "        -2 ${trimmedPath}/alignment/trimming/*${histName}*_R2*.fq.gz \\\n",
    "        -S ${outPath}/alignment/sam/${histName}_bowtie2_spikeIn.sam &> ${outPath}/alignment/sam/bowtie2_summary/${histName}_bowtie2_spikeIn.txt\n",
    "done\n",
    "\n",
    "##################################################################\n",
    "## Part 4. Calculate scale factor\n",
    "##################################################################\n",
    "ecoliPath=${projPath}/processed_data/CUT_RUN/Part2_ecoli_alignment/\n",
    "\n",
    "for histName in \"${histNames[@]}\"\n",
    "do\n",
    "\n",
    "    totalNum=$(sed -n '2p' $ecoliPath/alignment/sam/bowtie2_summary/${histName}_bowtie2_spikeIn.txt | awk '{print $1}')\n",
    "    ecoliNum=$(awk 'NR==4{sum+=$1} NR==5{sum+=$1; print sum}' $ecoliPath/alignment/sam/bowtie2_summary/${histName}_bowtie2_spikeIn.txt)\n",
    "    scale_factor=$(echo \"scale=4; $totalNum / $ecoliNum / 100\" | bc)\n",
    "    \n",
    "    {\n",
    "        echo $totalNum\n",
    "        echo $ecoliNum\n",
    "        echo $scale_factor\n",
    "    } > ${ecoliPath}/alignment/sam/bowtie2_summary/${histName}_bowtie2_scale_factor.txt\n",
    "    \n",
    "done\n",
    "\n",
    "date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1dcb43aa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "52b7092b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "deeptools",
   "language": "python",
   "name": "deeptools"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
