{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4dcd7759",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thu 18 Jan 2024 12:01:08 AM CST\n",
      "Thu 18 Jan 2024 01:37:51 AM CST\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "date\n",
    "\n",
    "########################################\n",
    "##### need to specify ################## \n",
    "projPath=/path/to/save/project/\n",
    "cores=4\n",
    "\n",
    "histNames=(C_H3K27me3 \\\n",
    "           C_H3K4me3 \\\n",
    "           C_IgG) \n",
    "\n",
    "histNames2=(C_H3K27me3 \\\n",
    "           C_H3K4me3) \n",
    "\n",
    "# Declare an associative array\n",
    "declare -A hist_control_map=([\"C_H3K27me3\"]=\"C_IgG\" \\\n",
    "                            [\"C_H3K4me3\"]=\"C_IgG\")\n",
    "########################################\n",
    "########################################\n",
    "\n",
    "##################################################################\n",
    "## Step 1. Preprocess files\n",
    "##################################################################\n",
    "\n",
    "inputPath=${projPath}/processed_data/CUT_RUN/Part1_alignment/\n",
    "inputPath2=${projPath}/processed_data/CUT_RUN/Part2_ecoli_alignment/\n",
    "outPath=${projPath}/processed_data/CUT_RUN/Part3_peak_calling/\n",
    "my_genome=${projPath}/meta/cut_run/sizes.genome\n",
    "\n",
    "## mkdir for output files\n",
    "mkdir -p $outPath/Step1_sortBam/sort_wo_n\n",
    "mkdir -p $outPath/Step2_bam2bedgraph/\n",
    "mkdir -p $outPath/Step3_peak_calling/\n",
    "\n",
    "for histName in \"${histNames[@]}\"\n",
    "do\n",
    " \n",
    "    scale_factor=$(sed -n '3p' $inputPath2/alignment/sam/bowtie2_summary/${histName}_bowtie2_scale_factor.txt | awk '{print $1}')\n",
    "    \n",
    "    bedtools bamtobed -bedpe -i $inputPath/alignment/bam/${histName}_bowtie2.mapped.bam \\\n",
    "        > $outPath/Step2_bam2bedgraph/${histName}.bed \\\n",
    "        2> $outPath/Step2_bam2bedgraph/${histName}.bed.log\n",
    "\n",
    "    awk '$1==$4 && $6-$2 < 1000 {print $0}' \\\n",
    "        $outPath/Step2_bam2bedgraph/${histName}.bed \\\n",
    "        > $outPath/Step2_bam2bedgraph/${histName}.clean.bed \\\n",
    "        2> $outPath/Step2_bam2bedgraph/${histName}.clean.bed.log\n",
    "\n",
    "    cut -f 1,2,6 $outPath/Step2_bam2bedgraph/${histName}.clean.bed | sort -k1,1 -k2,2n -k3,3n \\\n",
    "        > $outPath/Step2_bam2bedgraph/${histName}.fragments.bed\n",
    "\n",
    "    bedtools genomecov -bg -scale ${scale_factor} \\\n",
    "        -i $outPath/Step2_bam2bedgraph/${histName}.fragments.bed \\\n",
    "        -g $my_genome > $outPath/Step2_bam2bedgraph/${histName}.fragments.bedgraph \\\n",
    "        2> $outPath/Step2_bam2bedgraph/${histName}.fragments.bedgraph.log\n",
    "    \n",
    "    ## remove intermediate files\n",
    "    rm $outPath/Step2_bam2bedgraph/${histName}.fragments.bed\n",
    "    rm $outPath/Step2_bam2bedgraph/${histName}.clean.bed\n",
    "    rm $outPath/Step2_bam2bedgraph/${histName}.bed\n",
    "    \n",
    "done\n",
    "\n",
    "##################################################################\n",
    "## Step 2. Call peaks with SEACR\n",
    "##################################################################\n",
    "\n",
    "seacr=${projPath}/code/SEACR_1.3.sh\n",
    "\n",
    "for histName in \"${histNames2[@]}\"\n",
    "do\n",
    "    controlName=${hist_control_map[\"${histName}\"]}\n",
    "    \n",
    "    bash ${seacr} ${outPath}/Step2_bam2bedgraph/${histName}.fragments.bedgraph \\\n",
    "     ${outPath}/Step2_bam2bedgraph/${controlName}.fragments.bedgraph \\\n",
    "     non stringent ${outPath}/Step3_peak_calling/${histName}_seacr_${controlName}.peaks \\\n",
    "     &> ${outPath}/Step3_peak_calling/${histName}_seacr_${controlName}.peaks.log\n",
    "done\n",
    "\n",
    "\n",
    "date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed4b9283",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "deeptools",
   "language": "python",
   "name": "deeptools"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
